core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -129,12 +134,12 @@ public ProxyConfiguration(String name, int port, String userName, String passwor
         this(name,port,userName,password,noProxyHost,null);
     }
 
-    @DataBoundConstructor
     public ProxyConfiguration(String name, int port, String userName, String password, String noProxyHost, String testUrl) {
         this.name = Util.fixEmptyAndTrim(name);
         this.port = port;
         this.userName = Util.fixEmptyAndTrim(userName);
-        this.secretPassword = Secret.fromString(password);
+        String tempPassword = Util.fixEmptyAndTrim(password);
```

Comment:
Better hope actual passwords never have whitespace at the beginning or end.

Reply:
Yeah I generally do not recommend `trim`ming input unless you specifically know that leading/trailing whitespace could not be legal.